daemon/daemon.go

```diff
@@ -82,21 +79,22 @@ func (d *Daemon) Main(serve func(string, string)) int {
 // client.
 func (d *Daemon) Spawn() error {
 	binPath := d.BinPath
-	// Determine binPath.
-	if binPath == "" {
-		if len(os.Args) > 0 && path.IsAbs(os.Args[0]) {
-			binPath = os.Args[0]
-		} else {
-			// Find elvish in PATH
-			paths := strings.Split(os.Getenv("PATH"), ":")
-			result, err := util.Search(paths, "elvish")
-			if err != nil {
-				return errors.New("cannot find elvish: " + err.Error())
-			}
-			binPath = result
+	if !filepath.IsAbs(binPath) {
+		var err error
+		binPath, err = exec.LookPath(os.Args[0])
```

Comment:
The original code has the upside of not crashing when size of `os.Args` is 0 :).

Also, when Elvish is used as a login shell, `os.Args[0]` contains `-elvish`, and this code will search `-elvish` instead of `elvish`.

I think the original code is fine here...